src/pages/home/HeaderView.js

```diff
@@ -169,7 +169,7 @@ function HeaderView(props) {
         ),
     );
 
-    const canJoinOrLeave = isChatThread || !isSelfDM || isUserCreatedPolicyRoom || canLeaveRoom;
+    const canJoinOrLeave = (isChatThread || isUserCreatedPolicyRoom || canLeaveRoom) && !isSelfDM;
```

Comment:
Let's reformat for better readability, please: 
```suggestion
    const canJoinOrLeave = !isSelfDM && (isChatThread || isUserCreatedPolicyRoom || canLeaveRoom);
```